include/wx/dc.h

```diff
@@ -1364,6 +1366,9 @@ class WXDLLIMPEXP_CORE wxDC : public wxObject
 
     wxDCImpl * const m_pimpl;
 
+    void SetWindow(wxWindow* w)
```

Comment:
An alternative could be to get rid of `wxDCImpl::m_window` entirely and make `GetWindow()` virtual to allow reimplementing it properly in the derived classes (the base class version would just return `NULL`), including `wxBufferedPaintDC` that would return `m_paintdc.GetWindow()` from it.

I think this would be slightly more proper, but I could be missing some compatibility implications and it would, of course, be more work...